rzg2l-sbc: Support to build core-image-weston

khanhduydo-rvc opened the pull request:
Changes include:
    - Support for core-image-weston build
    - Support for building Mesa with Panfrost
    - Added user-defined weston-init config file

khanhduydo-rvc committed:
rzg2l-sbc: Support to build core-image-weston

Changes include:
    - Support for core-image-weston build
    - Support for building Mesa with Panfrost
    - Add graphic recipe and patches to implement rzg2l-du DRM entry points,
      so mesa can be used with it
    - Add packagegroup for gstreamer plugins
    - Add user-defined weston-init config file

Signed-off-by: Khanh Do Duy <khanh.do.yg@bp.renesas.com>
Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

khanhduydo-rvc committed:
rzg2l-sbc: Fix dependency loop error

Changes include:
    - Fix denpendency loop error when build image by modify index of IMAGE_FSTYPES variable

Signed-off-by: Khanh Do Duy <khanh.do.yg@bp.renesas.com>
Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

khanhduydo-rvc commented on conf/machine/rzg2l-sbc.conf:
`
ERROR: 5335 unbuildable tasks were found.########################################################
                             | ETA:  0:00:02
These are usually caused by circular dependencies and any circular dependency chains found will be printed below. Increase the debug level to see a list of unbuil
dable tasks.

Identifying dependency loops (this may take a short while)...

ERROR:
Dependency loop #1 found:
  Task /prj/soc-sw/local/work/khanh.do-duy/rzg_vlp_v3.0.6/poky/meta/recipes-core/images/core-image-initramfs-boot.bb:do_image_complete (dependent Tasks ['flash-wr
iter.bb:do_populate_sysroot', 'qemu-helper-native_1.0.bb:do_addto_recipe_sysroot', 'core-image-initramfs-boot.bb:do_image_cpio', 'qemu-system-native_9.0.2.bb:do_p
opulate_sysroot', 'core-image-initramfs-boot.bb:do_image_tar', 'firmware-pack.bb:do_populate_sysroot', 'trusted-firmware-a.bb:do_populate_sysroot', 'core-image-in
itramfs-boot.bb:do_image', 'core-image-initramfs-boot.bb:do_image_wic'])
  Task /prj/soc-sw/local/work/khanh.do-duy/rzg_vlp_v3.0.6/poky/meta/recipes-kernel/linux/linux-yocto_6.10.bb:do_bundle_initramfs (dependent Tasks ['core-image-ini
tramfs-boot.bb:do_image_complete', 'linux-yocto_6.10.bb:do_install'])
  Task /prj/soc-sw/local/work/khanh.do-duy/rzg_vlp_v3.0.6/poky/meta/recipes-kernel/linux/linux-yocto_6.10.bb:do_transform_bundled_initramfs (dependent Tasks ['lin
ux-yocto_6.10.bb:do_bundle_initramfs'])
  Task /prj/soc-sw/local/work/khanh.do-duy/rzg_vlp_v3.0.6/poky/meta/recipes-kernel/linux/linux-yocto_6.10.bb:do_deploy (dependent Tasks ['pigz_2.8.bb:do_populate_
sysroot', 'linux-yocto_6.10.bb:do_populate_sysroot', 'linux-yocto_6.10.bb:do_packagedata', 'linux-yocto_6.10.bb:do_bundle_initramfs', 'linux-yocto_6.10.bb:do_tran
sform_bundled_initramfs', 'depmodwrapper-cross_1.0.bb:do_populate_sysroot'])
  Task /prj/soc-sw/local/work/khanh.do-duy/rzg_vlp_v3.0.6/poky/meta/recipes-core/images/core-image-initramfs-boot.bb:do_image_wic (dependent Tasks ['core-image-in
itramfs-boot.bb:do_write_wks_template', 'mtools_4.0.44.bb:do_populate_sysroot', 'gptfdisk_1.0.10.bb:do_populate_sysroot', 'core-image-initramfs-boot.bb:do_image',
 'parted_3.6.bb:do_populate_sysroot', 'linux-yocto_6.10.bb:do_deploy', 'dosfstools_4.2.bb:do_populate_sysroot', 'core-image-initramfs-boot.bb:do_rootfs_wicenv'])

Dependency loop #2 found:
  Task /prj/soc-sw/local/work/khanh.do-duy/rzg_vlp_v3.0.6/poky/meta/recipes-core/images/core-image-initramfs-boot.bb:do_image_complete (dependent Tasks ['flash-wr
iter.bb:do_populate_sysroot', 'qemu-helper-native_1.0.bb:do_addto_recipe_sysroot', 'core-image-initramfs-boot.bb:do_image_cpio', 'qemu-system-native_9.0.2.bb:do_p
opulate_sysroot', 'core-image-initramfs-boot.bb:do_image_tar', 'firmware-pack.bb:do_populate_sysroot', 'trusted-firmware-a.bb:do_populate_sysroot', 'core-image-in
itramfs-boot.bb:do_image', 'core-image-initramfs-boot.bb:do_image_wic'])
  Task /prj/soc-sw/local/work/khanh.do-duy/rzg_vlp_v3.0.6/poky/meta/recipes-kernel/linux/linux-yocto_6.10.bb:do_bundle_initramfs (dependent Tasks ['core-image-ini
tramfs-boot.bb:do_image_complete', 'linux-yocto_6.10.bb:do_install'])
  Task /prj/soc-sw/local/work/khanh.do-duy/rzg_vlp_v3.0.6/poky/meta/recipes-kernel/linux/linux-yocto_6.10.bb:do_deploy (dependent Tasks ['pigz_2.8.bb:do_populate_
sysroot', 'linux-yocto_6.10.bb:do_populate_sysroot', 'linux-yocto_6.10.bb:do_packagedata', 'linux-yocto_6.10.bb:do_bundle_initramfs', 'linux-yocto_6.10.bb:do_tran
sform_bundled_initramfs', 'depmodwrapper-cross_1.0.bb:do_populate_sysroot'])
  Task /prj/soc-sw/local/work/khanh.do-duy/rzg_vlp_v3.0.6/poky/meta/recipes-core/images/core-image-initramfs-boot.bb:do_image_wic (dependent Tasks ['core-image-in
itramfs-boot.bb:do_write_wks_template', 'mtools_4.0.44.bb:do_populate_sysroot', 'gptfdisk_1.0.10.bb:do_populate_sysroot', 'core-image-initramfs-boot.bb:do_image',
 'parted_3.6.bb:do_populate_sysroot', 'linux-yocto_6.10.bb:do_deploy', 'dosfstools_4.2.bb:do_populate_sysroot', 'core-image-initramfs-boot.bb:do_rootfs_wicenv'])


ERROR: Command execution failed: 1

Summary: There were 3 ERROR messages, returning a non-zero exit code.
`
Error when enable initramfs feature. I'm fixing it.

khanhduydo-rvc replied:
Root cause: 
- When we define `IMAGE_FSTYPES:append = "wic.gz wic.bmap"` in **local.conf**, both the core-image-weston and core-image-initramfs-boot recipes use this variable. This leads to the creation of the core-image-initramfs-boot.bb:do_image_wic task, which results in a loop error (see the image).
![image](https://github.com/user-attachments/assets/1e497681-c214-43a7-9743-e19b9fbbd8c8)
- When we define `IMAGE_FSTYPES ?= "wic wic.gz wic.bmap"` in **rzg2l-sbc.conf**, only the core-image-weston recipe is affected. The core-image-initramfs-boot.bb:do_image_wic task is not created, resulting in **no error.**"

vudangRVC commented on .vscode/settings.json:
Can you remove this file? I think it's added by mistake?

vudangRVC commented on recipes-graphics/wayland/weston-init.bbappend:
Please ensure that the indentation is consistent.

tiennguyenuh commented:
@khanhduydo-rvc,

I built and tested your rootfs, and it seems like you're missing the entry point for DRI2: 

```
root@rzg2l-sbc:~# weston-simple-egl
MESA: error: ZINK: vkCreateInstance failed (VK_ERROR_INCOMPATIBLE_DRIVER)
libEGL warning: egl: failed to create dri2 screen
```

Please refer to my mesa recipe and see if it helps: https://github.com/vudangRVC/meta-renesas-sst/blob/styhead/rz-sbc-weston/recipes-graphics/mesa/mesa.bbappend

tiennguyenuh committed:
rzg2l-sbc: add linux patches and config fragment

Changes include:
    - Support the OV5640 camera and DA7219 audio codec.
    - Support Waveshare touch screen.
    - Enable clock and reset support for the WDT2 peripheral.
    - Fix a soft reset issue with Panfrost by restoring power
      domain properties.
    - Update configuration fragments to enable camera, audio
      and panfrost GPU support.
    - Add a V4L2 initialization script.

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh commented:
Hi @Renesas-SST/sst-rz,

The new update includes:
- Added a Linux patch to fix GPU soft reset issues triggered when using Panfrost.
- Added Linux patches and configuration fragments to enable audio and support for the OV5640 camera.
- Added v4l initilization script.

preetam-reddy commented on recipes-kernel/linux/rzg2l-sbc/dts-patches/0001-arm64-dts-renesas-Refactor-RZ-SBC-device-tree-and-re.patch:
```diff
@@ -1107,11 +1108,11 @@ index 000000000000..bbf478886312
-+        #sound-dai-cells = <0>;
-+        reg = <0x1a>;
-+    };
++	wm8978: codec@1a {
```
Check the indent here. Make sure its consistent and use tabs as per kernel guidelines..

preetam-reddy replied:
AS per @tiennguyenuh , Check patch was run on all patches.

preetam-reddy commented on recipes-kernel/linux/rzg2l-sbc/dts-patches/0006-drivers-gpu-panel-add-waveshare-panel-support-for-RZ.patch:
As this is an existing patch being ported, I'll skip review of it.

preetam-reddy commented on recipes-kernel/linux/rzg2l-sbc/dts-patches/0006-drivers-gpu-panel-add-waveshare-panel-support-for-RZ.patch:
```diff
@@ -0,0 +1,1752 @@
+ obj-$(CONFIG_DRM_PANEL_VISIONOX_RM69299) += panel-visionox-rm69299.o
+ obj-$(CONFIG_DRM_PANEL_VISIONOX_VTDR6130) += panel-visionox-vtdr6130.o
+ obj-$(CONFIG_DRM_PANEL_VISIONOX_R66451) += panel-visionox-r66451.o
+diff --git a/drivers/gpu/drm/panel/panel-toshiba-tc358762.c b/drivers/gpu/drm/panel/panel-toshiba-tc358762.c
```
Panel driver imported as is. No changes.

preetam-reddy commented on recipes-kernel/linux/rzg2l-sbc/dts-patches/0006-drivers-gpu-panel-add-waveshare-panel-support-for-RZ.patch:
```diff
@@ -0,0 +1,1752 @@
+ obj-$(CONFIG_UACCE)		+= uacce/
+ obj-$(CONFIG_XILINX_SDFEC)	+= xilinx_sdfec.o
+ obj-$(CONFIG_HISI_HIKEY_USB)	+= hisi_hikey_usb.o
++obj-$(CONFIG_RZSBC_MCU)        	+= rzsbc_mcu.o
```
Check alignment.

preetam-reddy commented on recipes-kernel/linux/rzg2l-sbc/dts-patches/0006-drivers-gpu-panel-add-waveshare-panel-support-for-RZ.patch:
```diff
@@ -0,0 +1,1752 @@
+ obj-$(CONFIG_NTSYNC)		+= ntsync.o
+ obj-$(CONFIG_HI6421V600_IRQ)	+= hi6421v600-irq.o
+ obj-$(CONFIG_OPEN_DICE)		+= open-dice.o
+diff --git a/drivers/misc/rzsbc_mcu.c b/drivers/misc/rzsbc_mcu.c
```
This is the backlight driver. It has some hooks that are exported to the touch driver.

preetam-reddy commented on recipes-multimedia/gstreamer/gstreamer1.0-plugins-bad_%d.bbappend:
```diff
@@ -0,0 +1 @@
+PACKAGECONFIG:append = " faac faad"
```
What is this?

vudangRVC replied:
Faad and Faac are GStreamer plugins used for audio encoding and decoding. These plugins serve as software-based encoders and decoders.
For more information, please check these URLs.
- https://gstreamer.freedesktop.org/documentation/[faac](https://gstreamer.freedesktop.org/documentation/faac/index.html?gi-language=c)/index.html?gi-language=c
- https://gstreamer.freedesktop.org/documentation/[faad](https://gstreamer.freedesktop.org/documentation/faad/index.html?gi-language=c#faad-page)/index.html?gi-language=c#faad-page
